transcom/mymove
# INT B-21485 SC state management to avoid duplicate moves

> [!IMPORTANT]  
> This PR contains the code from [B-20976](https://github.com/transcom/mymove/pull/13788) and is essentially downstream of it. Please ensure this PR is merged AFTER that, but it can be tested prior to it being merged.

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21485)

## Summary

We only want the services counselors to only add new orders information for a customer when they start from the Create New Move flow after searching for a customer. This is an effort to prevent a SC from accidentally creating duplicate moves for a customer by clicking the browser back button after adding orders information.

This PR does the following:
- updates `canAddOrders` to `true` when the SC clicks the create new move button
- updates `canAddOrders` to `false` on home page load & after submitting orders information
- adds/updates relevant tests

### How to test

1. Access app as an office user and log in as a services counselor
2. Go to the customer search and search for an existing user
3. Click on the "Create a move" button that appears
4. Update the customer's profile and get to the add orders form
5. Submit the orders info and you should see the move details page
6. Hit the browser back button - you should be navigated to the SC queue instead of the add orders page

## Screenshots
When searching for customers, `canAddOrders` will be `false`
![Screenshot 2024-09-30 at 12 17 00 PM](https://github.com/user-attachments/assets/35c0f1b0-d1b3-4c65-a74f-ec4efd2a8fe0)

Once they click the create new move button, that value is being changed to `true` - it will remain true throughout the entire flow all the way up until they add/submit the orders info
![Screenshot 2024-09-30 at 12 17 13 PM](https://github.com/user-attachments/assets/a4033b2c-1284-437d-b72a-2a75ca9b735e)

Once they submit the orders/create the move, `canAddOrders` will be changed to `false`. If the SC user tries to access the add orders route and `canAddOrders` is `false`, they will be redirected to their base queue.
![Screenshot 2024-09-30 at 12 17 50 PM](https://github.com/user-attachments/assets/8ded26be-1262-40b4-89a4-e3bc436f2dbc)
